Lots of new stuffs :D

antonleviathan opened the pull request.

antonleviathan committed:
move petrinet.js into app.js and create _includes which add scripts and
head to default layout

antonleviathan committed:
minor formatting and cleanup

antonleviathan committed:
add detail pages for all nets

antonleviathan committed:
minor styling of index.md content

wires commented on _includes/scripts.html:
```diff
@@ -0,0 +1,132 @@
+<script src="js/lodash.min.js"></script>
+<script src="js/app.js"></script>
+
+<script>
```
Remove this `<script>` and add `<script>` tags in the markdown that define the net and call the "render" function

antonleviathan replied:
Done.

wires commented on _layouts/details.html:
```diff
@@ -0,0 +1,26 @@
+        </div>
+        <div class="vert-align mdl-cell mdl-cell--6-col mdl-cell--8-col-tablet">
+            <!-- description - lorem ipsum -->
+            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
```
put `{{ content }}` here and remove below

antonleviathan replied:
Done.

wires commented on producer-consumer.md:
```diff
@@ -0,0 +1,6 @@
+layout: details 
+title: Producer - Consumer 
+main_title: Producer - Consumer 
+pnet: pnet1
```
instead of referring to the net in frontmatter, directly declare the net in `script` tag in the markdown

antonleviathan replied:
Done.

wires commented on producer-consumer.md:
```diff
@@ -0,0 +1,6 @@
+---
+layout: details 
+title: Producer - Consumer 
+main_title: Producer - Consumer 
```
remove `main_title`, and then use `h1` in the markdown to give titles

antonleviathan replied:
Done.

wires commented on _layouts/details.html:
```diff
@@ -0,0 +1,26 @@
+<body id="top">
+    <div class="title-cont">
+        <div class="inner">
+            {{ page.main_title }}
```
this should go and be done in the markdown file

antonleviathan replied:
Done.

wires commented on _layouts/details.html:
```diff
@@ -0,0 +1,26 @@
+        </div>
+    </div>
+    <div class="content-grid mdl-grid nets-container">
+        <div id="{{page.pnet}}" class="vert-align text-center mdl-cell mdl-cell--6-col mdl-cell--8-col-tablet">
```
same here, this should be done in the markdown file as:

```
<script>
const net = ....
renderNet(net); // like drawNet but inserts a div with id (document.write ...)
</script>
```

antonleviathan committed:
integrate jelle's feedback - move pnet scripts into md files and add
navbar

antonleviathan commented:
I added most of the changes and also introduced a new navbar. Wasn't sure how to go about implementing Pawel's pnet. Will talk to you in person about this. @wires

wires commented on traffic-lights.md:
```diff
@@ -0,0 +1,59 @@
+    scaleModel(transitions_pnet3, states_pnet3, 4, 4);
+
+    // drawing stuffs
+    drawNet('#pnet3', states_pnet3, transitions_pnet3, marking_pnet3);
```
okay, lets add one more change, which is to wrap this into a function:

```js
function insertNet(states, transitions, marking, scale) {
  const id = 'ptnet-' + Math.random().toString()
  document.write(`<div id="${id}"></div>`)
  scaleModel(transitions, states, scale, scale)
  drawNet(`#${id}`, states, transitions, marking)
}
```

then you can remove that code from these script tags

wires reviewed (changes requested):
Great, can't wait to merge this 💃

antonleviathan committed:
wrap drawNet and scale functions to clean up detailed net pages

antonleviathan committed:
fix some container nesting and minor cleanup

antonleviathan commented:
I think we're good to merge. We do need to update the content as well though but that can happen over time I guess. @wires

wires commented on producer-consumer.md:
```diff
@@ -0,0 +1,46 @@
+
+this is an importatnt net becaus ahhahaha
+
+[edit](https://github.com/PetriNets/petrinets.github.io/edit/master/producer-consumer.md)
```
This should be done in the template `<a href="{{site.github.repository_url}}/blob/gh-pages/{{page.path}}">Edit this page</a>`, see [this comment](https://github.com/jekyll/jekyll-help/issues/5#issuecomment-39033862), but will merge it for now
